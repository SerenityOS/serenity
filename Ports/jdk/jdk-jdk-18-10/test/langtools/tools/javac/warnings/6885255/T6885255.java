/**
 * @test /nodynamiccopyright/
 * @bug 6885255
 * @summary -Xlint:rawtypes
 * @compile/ref=T6885255.out -XDrawDiagnostics -Xlint:rawtypes T6885255.java
 */

class T6885255 {

    static class Test<X, Y> {}

    Class<Test> ct; //no warn - outer Class w/ raw param
    Class<Test<Test, Test>> ctt; //warn - outer Class w/o raw param (2)

    Class<Class<Test>> cct; //warn - outer Class w/o raw param
    Class<Class<Test<Test, Test>>> cctt; //warn - outer Class w/o raw param (2)

    Object o1 = (Test)null; //no warn - outer raw and cast
    Object o2 = (Test<Test, Test>)null; //warn - inner raw (2)

    Object o3 = (Class)null; //no warn - outer raw and cast
    Object o4 = (Class<Test>)null; //no warn - outer Class w/ raw param

    Object o5 = (Class<Test<Test, Test>>)null; //warn - outer Class w/ non raw param (2)
    Object o6 = (Class<Class<Test<Test, Test>>>)null; //warn - outer Class w/ non raw param (2)

    Object o7 = (Test<Class, Class>)null; //warn - inner raw (2)
    Object o8 = (Test<Class<Test>, Class<Test>>)null; //warn - inner Class (2)

    boolean b = null instanceof Test; //no warn - raw and instanceof
}
