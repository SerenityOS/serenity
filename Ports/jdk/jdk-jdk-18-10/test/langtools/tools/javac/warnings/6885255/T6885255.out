T6885255.java:13:16: compiler.warn.raw.class.use: T6885255.Test, T6885255.Test<X,Y>
T6885255.java:13:22: compiler.warn.raw.class.use: T6885255.Test, T6885255.Test<X,Y>
T6885255.java:15:17: compiler.warn.raw.class.use: T6885255.Test, T6885255.Test<X,Y>
T6885255.java:16:22: compiler.warn.raw.class.use: T6885255.Test, T6885255.Test<X,Y>
T6885255.java:16:28: compiler.warn.raw.class.use: T6885255.Test, T6885255.Test<X,Y>
T6885255.java:19:23: compiler.warn.raw.class.use: T6885255.Test, T6885255.Test<X,Y>
T6885255.java:19:29: compiler.warn.raw.class.use: T6885255.Test, T6885255.Test<X,Y>
T6885255.java:24:29: compiler.warn.raw.class.use: T6885255.Test, T6885255.Test<X,Y>
T6885255.java:24:35: compiler.warn.raw.class.use: T6885255.Test, T6885255.Test<X,Y>
T6885255.java:25:35: compiler.warn.raw.class.use: T6885255.Test, T6885255.Test<X,Y>
T6885255.java:25:41: compiler.warn.raw.class.use: T6885255.Test, T6885255.Test<X,Y>
T6885255.java:27:23: compiler.warn.raw.class.use: java.lang.Class, java.lang.Class<T>
T6885255.java:27:30: compiler.warn.raw.class.use: java.lang.Class, java.lang.Class<T>
T6885255.java:28:29: compiler.warn.raw.class.use: T6885255.Test, T6885255.Test<X,Y>
T6885255.java:28:42: compiler.warn.raw.class.use: T6885255.Test, T6885255.Test<X,Y>
15 warnings
